interpreter.js

```diff
@@ -2808,6 +2808,10 @@ Interpreter.prototype['stepCallExpression'] = function() {
     state.doneExec_ = true;
     var func = state.func_;
     if (!func || !func.isObject) {
+      var components = state.components_;
+      if (components && Array.isArray(components) && 1 <= components.length) {
+        func = node['callee'].object.name + '.' + components[1];
```

Comment:
it's not clear to me why this is guaranteed to do what we want. Specifically, what about the scenario where `func` is truthy, but `func.isObject` is falsy?

Are you also certain it's okay to redefine `func` here, given that we use it later in this method?

Reply:
looks like this introduces problems for the following test case:

```js
var foo;
foo()
```

Reply:
You made three points:
1. Great point. I've split up the functionality to account for this.
2. Should be fine to redefine func since all paths lead to a throw. I changed it anyway for readability in v2.
3. Thanks for catching that! Fixed in the second iteration.